gnosis/verification.k

```diff
@@ -72,22 +72,52 @@ module VERIFICATION
 
 // Reading symbolic WordStack from memory, then extracting bytes from it. General case (stack size can be != 32)
 //========================================================================
-    rule nthbyteof(chop(#asWord(W : WS) &Int MASK), I       , N       )
-      => nthbyteof(chop(#asWord(    WS) &Int MASK), I -Int 1, N -Int 1)
-      requires I >Int 0
+    //decoding nthbyteof(#asWord())
+    rule nthbyteof(#asWord(W : WS), I       , N       )
+      => nthbyteof(#asWord(    WS), I -Int 1, N -Int 1)
+      requires I >Int 0 andBool #sizeWordStack(W : WS) <=Int 32
 
-    rule nthbyteof(chop(#asWord(W : WS) &Int MASK), 0, N) => W &Int ((MASK >>Byte #sizeWordStack(WS)) &Int 255)
+    rule nthbyteof(#asWord(W : WS), 0, N) => W
+      requires #sizeWordStack(W : WS) <=Int 32
```

Comment:
i think it should be `#sizeWordStack(W : WS) <=Int N`

Reply:
Good catch! Actually it should be `#sizeWordStack(W : WS) ==Int N andBool N <=Int 32` If it's strictly `#sizeWordStack(W : WS) <Int N` then wrong byte will be selected.

Will correct here and below.